src/contracts/examples/intents-example/SwapIntentInvertBidDAppControl.sol

```diff
@@ -87,14 +87,8 @@ contract SwapIntentInvertBidDAppControl is DAppControl {
         require(msg.sender == ATLAS, "SwapIntentDAppControl: InvalidSender");
         require(address(this) != CONTROL, "SwapIntentDAppControl: MustBeDelegated");
 
-        address user = _user();
-
-        require(
-            _availableFundsERC20(
-                swapIntent.tokenUserSells, user, swapIntent.maxAmountUserSells, ExecutionPhase.PreSolver
-            ),
-            "SwapIntentDAppControl: SellFundsUnavailable"
-        );
+        // Transfer to the Execution Environment the amount that the solver is invert bidding
+        _transferUserERC20(swapIntent.tokenUserSells, address(this), swapIntent.maxAmountUserSells);
```

Comment:
I think this works, so this transfer into the EE (rather than directly to a solver) I believe is solely to ensure correct bid amount calculation by the current logic? as the current logic checks the balance of the EE directly before the preSolver delegateCall.

Reply:
Yes, we need to do it this way for the bid tracking in EE.